modules/core/src/test/java/org/apache/ignite/failure/FailureHandlerLoggingTest.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2019 GridGain Systems, Inc. and Contributors.
+ *
+ * Licensed under the GridGain Community Edition License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.gridgain.com/products/software/community-edition/gridgain-community-edition-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.failure;
+
+import java.util.Collections;
+import java.util.HashSet;
+import org.apache.ignite.internal.IgniteEx;
+import org.apache.ignite.internal.processors.failure.FailureProcessor;
+import org.apache.ignite.testframework.ListeningTestLogger;
+import org.apache.ignite.testframework.LogListener;
+import org.apache.ignite.testframework.junits.common.GridCommonAbstractTest;
+import org.junit.Test;
+
+/**
+ * Tests log messages of failure handler.
+ */
+public class FailureHandlerLoggingTest extends GridCommonAbstractTest {
+    /**
+     * Logger for listen messages.
+     */
+    private final ListeningTestLogger log = new ListeningTestLogger(false, GridCommonAbstractTest.log);
+
+    /** {@inheritDoc} */
+    @Override protected void afterTest() throws Exception {
+        super.afterTest();
+
+        log.clearListeners();
+
+        stopAllGrids();
```

Comment:
You can use the same Ignite instance for both test cases. Just start node before all tests and stop it after all tests. Starting of node is time consuming operation.